`./tools` folder fails to build with unit tests that are never triggered

**What happened**:
As @akalenyu mentions in [this comment](https://github.com/kubevirt/kubevirt/pull/12778#discussion_r1751678770), `./tools` fails to build:
```
kubevirt/tools/util/marshaller_test.go:32:195: not enough arguments in call to components.NewHandlerDaemonSet
```

In addition, as he's pointing out, these tests are not attempted to ever run:
https://github.com/kubevirt/kubevirt/blob/941b04cb9f90745ae2038b13c8bb82eea55a16dd/hack/bazel-test.sh#L7

To add an insult to injury, these are lacking the `*_suite_test.go` that's required to make these tests runnable.

**What you expected to happen**:
The tests should either be able to build and run via `make test` or removed.

**How to reproduce it (as minimally and precisely as possible)**:
Trying to build `./tools`

--- a/hack/bazel-test.sh
+++ b/hack/bazel-test.sh
@@ -4,8 +4,7 @@ source hack/common.sh
 source hack/bootstrap.sh
 source hack/config.sh
 
-WHAT=${WHAT:-"//staging/src/kubevirt.io/... //pkg/... //cmd/... //tests/framework/..."}
-
+WHAT=${WHAT:-"//staging/src/kubevirt.io/... //pkg/... //cmd/... //tools/util/... //tools/cache/... //tests/framework/..."}
 rm -rf ${ARTIFACTS}/junit ${ARTIFACTS}/testlogs
 
 if [ "${CI}" == "true" ]; then

--- a/tools/util/marshaller_test.go
+++ b/tools/util/marshaller_test.go
@@ -29,11 +29,7 @@ import (
 
 func TestMarshallObject(t *testing.T) {
 	var imagePullSecret []v1.LocalObjectReference
-	handler, err := components.NewHandlerDaemonSet("{{.Namespace}}", "", "{{.DockerPrefix}}", "{{.DockerTag}}", "", "", "", "", "", "", "", "", v1.PullIfNotPresent, imagePullSecret, nil, "2", nil, false)
-	if err != nil {
-		t.Fatalf("error generating virt-handler deployment for marshall test %v", err)
-	}
-
+	handler := components.NewHandlerDaemonSet("{{.Namespace}}", "", "{{.DockerPrefix}}", "{{.DockerTag}}", "", "", "", "", "", "", "", "", "", "", v1.PullIfNotPresent, imagePullSecret, nil, "2", nil, false)
 	writer := strings.Builder{}
 
 	MarshallObject(handler, &writer)

